First draft of the SLA (#15)
CSCfi/rahti-user-guide: docs/introduction/sla.md

# Rahti Container Cloud Service Service Level Agreement DRAFT

## General

This Service Level Agreement (hereafter called SLA) is made between the
customer, Rahti Container Cloud Service and the service provider, CSC - IT
Center for Science Ltd., to cover the provision and support of the service as
described hereafter. Amendments, comments and suggestions must be addressed

> using the communication channels defined in section 8.1. The service provider

Reviewer: Here we can't refer to section 8.1 as vanilla markdown doesn't have numbered headings. A reference to the correct heading could be used instead.

Author: Fixed all the references to markdown references

Reviewer: It looks like there's still two references in the text to numbered section. You can find these by searching for "section 6".

Author: Fixed

retains right to introduce changes to the infrastructure. If the Customer does
not accept the changes, this service subscription can be terminated.

## Scope and description of the service

> This SLA applies to Rahti Container Cloud Service1 (hereafter referred to as

Reviewer: ```suggestion
This SLA applies to the Rahti Container Cloud Service (hereafter referred to as
```

Author: Corrected working and typos

Rahti or the Rahti service). The Rahti service is a cloud computing service.
The Rahti service is based on OKD, which is a distribution of Kubernetes. The
Rahti serviceallows its customers or users to run their own workloads on top of

> it, often implemented as containers.  The Rahti service provides, but is not

Reviewer: ```suggestion
it, often implemented as containers. The Rahti service provides, but is not
```

Author: Fixed spaces across document, might need consistent line changes though

 * Storage services 
 * Virtual networks for connecting container instances
 * Load balancing of traffic to user workloads
 * Features for replication, rolling updates, auto-recovery and auto-scaling of user workloads
 * Basic default domain name and TLS for hosted web applications (under rahtiapp.fi)

Customers can manage their resources using a web interface accessible through a
web browser and through a set of APIs which allow programmatic management of
resources. In order to access and use the service the customer must have a CSC
user account. The workloads are separated from other customers’ workloads from
a network, storage and computational view.

This service is provided under User policy and terms of usage for the Rahti
contariner cloud service hence by agreeing to this document the customer also
agrees with CSC General Terms of Use and CSC’s Security Policy.

## Service hours & exceptions

The service is designed to run continuously. However, the following exceptions apply:

Scheduled maintenances:

 * Planned service breaks - Announced to the customer at least three weeks in advance
 * Unplanned service breaks

## Service components & dependencies

> The service covered by this SLA is made up of the following (technical and logical) service components:

Reviewer: The list below lists service dependencies, but these are labeled as service components. Service components are presumably different from these. Does FitSM have some sort of a clear definition for what constitutes a service dependency and what constitutes a component? I.e. where do we draw the borders of what is part of the service itself and what are external systems and infrastructure that it depends on.

I also noticed that the content below this subheading is currently the same as in the cPouta SLA.

Author: I read the FitSM SLA template, and changed this based my reading of the template. https://fitsm.itemo.org/fitsm-templates-samples-and-guides/
Now it only has services at a very high level.

Phone |  +358 (0) 94 57 2821
E-Mail | servicedesk@csc.fi
Webpage and contact form | https://research.csc.fi/cloud-computing https://research.csc.fi/support
Response time target | Within Three Working Days

###  Incident handling

Disruptions to the agreed service functionality or quality will be handled
according to an appropriate priority based on the impact and urgency of the
incident. In this context, the following general priority guidelines apply:

1. Ensuring normal levels of security
2. Restoring normal service operation
3. Restoring customer workloads where possible

Response and resolution times are provided as service level targets (see section 6).

### Fulfilment of service requests

In addition to resolving incidents, the following standard service requests are
defined and will be fulfilled through the defined support channels:

 * Issues regarding CSC’s identity management service
 * Issues using this service’s web interface
 * Issues using the provided APIs
 * Issues deploying workloads
 * Issues using workload templates

Response and fulfilment times are provided as service level targets (see section 6).

## Service level targets

> The Rahti service level targets adhere to JHS1745 as follows:

Reviewer: ```suggestion
The Rahti service level targets adhere to JHS 174 as follows:
```

Author: fixed

planned mitigation actions and expected resolution time.

## Limitations and constraints

The provisioning of the service under the agreed service level targets is subject to the following limitations and constraints: 

 * Support is provided primarily in the following languages: English, Finnish
 * Downtimes caused due to upgrades for fixing critical security issues are not considered SLA violations. In the case of critical security upgrades CSC reserves the right to apply the upgrades with minimal notice
 * A technical failure which affects individual customers does not count as downtime
 * Failures of single instances of the workload (containers) are expected behaviour, and does not count as downtime
 * Any other causes outside service provider’s direct control

## Communication, reporting and escalation

### General communication

The following contacts will be generally used for communications related to the
service in the scope of this SLA:

| Communication channels ||
--- | ---
Contact for Customers | servicedesk@csc.fi
Security incident reporting | (CSC’s Head of Security) security@csc.fi +358 (0) 94 57 2253

### Reporting

> The service availability information is…. The information provided will be limited to service availability and by data security and privacy constraints.

Reviewer: "The service availability information is…" Seems to be missing some text?

Author: I added a statement that the availability report is available through a request to servicedesk

### Escalation and complaints

For escalation and complaints, the defined contact point shall be used, and the
following rules apply:

> 1. First contact shall be established, preferably by e-Mail, to Contact for Customers address (See section 8.1) explaining the reason for the complaint with a sensible level of detail and clarity. Please also include, if possible, the following information:

Reviewer: There's a reference to section 8.1, but that doesn't exist in the markdown document.

Author: Fixed all the references to markdown references

    * Name of the service
    * Date and time of the events
    * Usernames of affected users
    * Channel to use on following communications (If other is preferred)
2. CSC Service Desk will contact you back within three working days with information about the incident and which procedures will be adopted. 

## Information security & data protection
 CSC has approved a security policy and also follows security best practices.
For CSC's customers, partners and staff there are detailed security guidelines.
Many items in our security policies and guidelines refer to external compliance
requirements. CSC also has procedures for risk and security management. For
more information, please refer to the following pages:

https://www.csc.fi/security

> https://www.csc.fi/web/research/rahti-security

Reviewer: The Rahti security guidelines will be under the rahti.csc.fi domain and later under the docs.csc.fi domain.

Author: We'll have to go through all links to the Rahti docs. As afaik we have no Rahti security guidelines right now,  we can remove the link altogether.

Reviewer: It's currently a work in progress by @juhanikataja . It might be that the security guidelines are finished in time for the SLA.

The handling of personal information on the Rahti service is described here.

> https://www…..

Reviewer: This could be under rahti.csc.fi as well if the SLA is there.

Author: I think we need to have a rahti service page under research.csc.fi to (as research.csc.fi/cpouta) to be consistent across our services. That's probably a good canonical place for at least links to all the important docs.

## Additional responsibilities of the Service Provider

Additional responsibilities of the Service Provider are as follow:

>  * Adhere to all applicable operational and security policies and procedures defined in CSC’s Security Policy (See section 9) and to other policy documents referenced therein

Reviewer: Reference to "section 9" should be changed.

Author: Fixed all the references to markdown references

 * Use communication channels defined in the agreement
 * Provide monitoring data to measure fulfillment of agreed service level targets.

## Customer responsibilities

> The Customer agrees to follow the General Terms of Use for CSC's Services for

Reviewer: The references (7,8,9) in the paragraph could be converted to links instead.

Author: Fixed all the references to markdown references

> Science7, User policy and terms of usage for the Rahti container cloud service8

Reviewer: ```suggestion
Research, User policy and terms of use for the Rahti container cloud service
```

Author: fixed wording

Reviewer: The general ToU document is called the "General Terms of Use for CSC's Services for **Research**". In the current version of this SLA document it is called the "General Terms of Use for CSC's Services for **Science**".

Reviewer: The Rahti Tou is currently called the "User policy and terms of use for the Rahti container cloud service". The current version of this SLA refers to it as "Terms of use for the Rahti container cloud service".

Author: Should be correct now

Reviewer: I'll add to this existing review so as to avoid opening a new one. There's another place in the document under "Scope and description of the service" where it says "User policy and terms of usage for the Rahti
contariner cloud service" which could also be corrected to "User policy and terms of use for the Rahti container cloud service" (i.e. usage -> use, contariner -> container).

Author: I removed the "Scope and description of the service" mention of the TOU and security requirements as they are already mentioned under the user responsibilities.

and CSC’s Security Policy9.

## Review

There will be reviews of the service performance against service level targets
and of this SLA at planned intervals according to the following rules: 

>  * Annual reviews are done internally and based on customer feedback

Reviewer: ```suggestion
 * Annual reviews are done internally and based on customer feedback.
```

>  *  Major changes to the service may trigger a review.

Reviewer: ```suggestion
 * Major changes to the service may trigger a review.
```

> ## Glossary of terms

Reviewer: Any necessary terms from the ToU document also used in this document could be listed here as well.

Author: Added generic relevant terms from the ToU, and tried to fix the wording across the document to match.

Reviewer: Could you also make the usage of emphasis the same across the items in the glossary list? Some items now have emphasis while others do not.

Author: Yes. Made

user account.

The user's applications are separated from other users’ applications from a
network, storage and computational view.

The Rahti service accounts for service usage based on the resources used. Up to
date resource cost information is found in the [Rahti documentation](https://rahti.csc.fi)

The normal functioning of the service is defined as

 * the ability to launch new user applications.
 * the availablility and integrity of the storage services.
 * the functioning of the network, and the accessibility of user applications.
 * the functioning of replication, auto-recovery and auto-scaling of user applications.

This service is provided under User policy and terms of usage for the Rahti
contariner cloud service hence by agreeing to this document the customer also
agrees with CSC General Terms of Use and CSC’s Security Policy.

## Service hours & exceptions

The service is designed to run continuously. However, the following exceptions apply:

Scheduled maintenances:

 * Planned service breaks - Announced to the customer at least three weeks in advance
 * Unplanned service breaks

## Service components & dependencies

> The Rahti service is depends on the following services. The dependencies of these services are described in their respective service documentation.

Reviewer: ```suggestion
The Rahti service depends on the following services. The dependencies of these services are described in their respective service documentation.
```

 * The cPouta service

>  * CSC user administration

Reviewer: This is "CSC’s identity management service" in the cPouta SLA.

Author: Correct. However I think CSC user management might describe the dependency better, since it also depends on all the processes.

Reviewer: Should we also link to the documentation here? The text says that "The dependencies of these services are described in their respective service documentation.", but as a prospective user of Rahti, would I know where to find this documentation without links? Would I be able to find a document that describes a service called "CSC user administration"? If that document exists, would it be easily identifiable as the service documentation for "CSC user administration", or would it have a different title?

It might be that not all of the dependencies have such documentation, or it would not be relevant. For example, the link for the service desk could simply be https://research.csc.fi/support that has a contact form and information about the service desk.

Listing "Kajaani datacenter infrastructure" and "Funet network" here explicitly could also be helpful, as customers might not otherwise realize that Rahti is dependent on them. This could be relevant in a situation where Rahti is unavailable due to a problem in the Funet network, for example. While cPouta has these in its SLA already and cPouta is listed here, the connection may not be obvious to end users.

Author: Added the other services as references, but removed the statement of their dependency lists. I don't see that we'd like to maintain that list in the Rahti SLA

>  * CSC Servicedesk for user support

Reviewer: This is "CSC’s helpdesk" in the cPouta SLA, though the wording here might be better.

## Support

> Support to the services covered by the scope of this SLA are provided through

Reviewer: ```suggestion
Support for the services covered by the scope of this SLA are provided through
```
